apps/meteor/app/custom-oauth/server/custom_oauth_server.js

```diff
@@ -349,7 +349,8 @@ export class CustomOAuth {
 					user.services &&
 					user.services[serviceName] &&
 					user.services[serviceName].id === serviceData.id &&
-					user.name === serviceData.name
+					user.name === serviceData.name &&
+					user.email === serviceData.email
```

Comment:
```suggestion
					(this.keyField === 'email' || user.emails?.find(({address}) => address === serviceData.email))
```
`user.email` doesn't exist.